pkg/venafi/cloud/connector.go

```diff
@@ -775,42 +775,54 @@ func (c *Connector) ProvisionCertificate(req *endpoint.ProvisioningRequest, opti
 	log.Println("Certificate is valid for provisioning (VCP generated)")
 
 	// setting options for provisioning
-	log.Println("setting provisioning options")
-	provisioningOptions, err := setProvisioningOptions(options)
-	if err != nil {
-		return nil, err
+	var provisioningOptions *cloudproviders.CertificateProvisioningOptionsInput
+	if options != nil {
+		log.Println("setting provisioning options")
+		provisioningOptions, err = setProvisioningOptions(options)
+		if err != nil {
+			return nil, err
+		}
+		log.Println("provisioning options successfully set")
 	}
-	log.Println("provisioning options successfully set")
 
-	if reqData.KeystoreID == nil {
-		if reqData.ProviderName == nil || reqData.KeystoreName == nil {
-			return nil, fmt.Errorf("any of keystore ID or both Provider Name and Keystore Name must be provided for provisioning")
+	ctx := context.Background()
+
+	var keystoreIDString string
+	var cloudKeystoreType string
+
+	if reqData.Keystore == nil {
+		if reqData.KeystoreID == nil {
+			if reqData.ProviderName == nil || reqData.KeystoreName == nil {
+				return nil, fmt.Errorf("any of keystore ID or both Provider Name and Keystore Name must be provided for provisioning")
```

Comment:
Check if this error message needs adjusting (since there is `reqData.Keystore` check now), resolve here if not.

Reply:
Fixed by latest changes